guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,446 @@
+```
+
+From inside the `wasmd` directory, run the init script:
+
+```bash
+curl -sSL https://rollkit.dev/cosmwasm/init.sh | perl -pe 's/127\.0\.0\.1/0.0.0.0/g' | sh
+```
+
+:::info
+We modify the listening address from `127.0.0.1` to `0.0.0.0` so that it is accessible from the Hyperlane docker agents later on.
+:::
+
+With that, we have kickstarted our `wasmd` network!
+
+## 💻 Deploy the Hyperlane contracts {#deploy-contracts}
+
+Fork the cw-hyperlane repo:
+
+```bash
+git clone --depth 1 git@github.com:many-things/cw-hyperlane.git
+cd cw-hyperlane
+```
+
+#### Create `config.yaml` with our networks setup:
+
+```bash
+echo 'networks:
+  - id: "localwasmd"
+    hrp: "wasm"
+    endpoint:
+      rpc: "http://localhost:36657"
+      rest: "http://localhost:1317"
+      grpc: "http://172.17.0.1:9290"
+    gas:
+      price: "0.025"
+      denom: "uwasm"
+    domain: 1063
+  - id: "stride-internal-1"
+    hrp: "stride"
+    endpoint:
+      rpc: "https://stride.testnet-1.stridenet.co"
+      rest: "https://stride.testnet-1.stridenet.co/api"
+      grpc: "http://stride-testnet-grpc.polkachu.com:12290"
+    gas:
+      price: "0"
+      denom: "ustrd"
+    domain: 1651
+
+# wasm133xh839fjn9wxzg6vhc0370lcem8939zr8uu45
+# stride133xh839fjn9wxzg6vhc0370lcem8939z2sd4gn
+signer: "join always addict position jungle jeans bus govern crack huge photo purse famous live velvet virtual weekend hire cricket media dignity wait load mercy"
+
+deploy:
+  ism:
+    type: routing
+    owner: <signer>
+    isms:
+      - type: multisig
+        owner: <signer>
+        validators:
+          1651:
+            addrs:
+              - <signer>
+            threshold: 1
+  hooks:
+    default:
+      type: aggregate
+      owner: <signer>
+      hooks:
+        - type: merkle
+
+        - type: igp
+          owner: <signer>
+          configs:
+            1651:
+              exchange_rate: 2
+              gas_price: 5000
+          default_gas_usage: 100000
+
+    required:
+      type: aggregate
+      owner: <signer>
+      hooks:
+        - type: pausable
+          owner: <signer>
+          paused: false
+
+        - type: fee
+          owner: <signer>
+          fee:
+            # defaults tp gas denom from network config
+            denom: uwasm
+            amount: 1' > config.yaml
+```
+
+:::tip
+The `domain` IDs were generated as a sum or the ascii values of the chain ID, for example:
+
+```python
+print(sum(ord(char) for char in "stride-internal-1"))
+```
+
+However, you can pick any number as your domain ID.
+:::
+
+:::tip
+In this guide, we're using a predefined seep phrase for simplicity. Alternativley you can generate and use your own:
+
+```bash
+wasmd keys add my-key
+```
+
+To get your Stride address, import the generated seed phrase to the `strided` CLI using:
+
+```bash
+strided keys add --recover my-key
+```
+
+Then fund it using the [Stride testnet faucet](https://stride-faucet.pages.dev).
```

Comment:
do you have rate limits on this?

Reply:
No, why? I don't think we need to rate limit this. There's not a lot of funds there and it's all client side, no backend

Reply:
can't someone just drain it?

Reply:
yes, but it's testnet. the next one will complain and we'll top it up or find a better way to do it, but for now that's the easiest way